src/components/NewCommentForm.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import { Paper } from "@mui/material";
+import { postNewComment } from "../api";
+import { useParams } from "react-router-dom";
+
+const NewCommentForm = ({ comments, setComments }) => {
+  const { article_id } = useParams();
+  const [name, setName] = useState("");
+  const [body, setBody] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setComments(() => {
+      return [
+        { author: name, body: body, votes: 0, created_at: "Just now" },
+        ...comments,
+      ];
+    });
+    postNewComment(article_id, name, body).then((data) => {
```

Comment:
If this request goes wrong - for example if I use an invalid username, like "Katherine", currently the state updates and the comment appears on the page as if it was successful, but the comment disappears upon page refreshing.
We need a better way of letting the user know if their request hasn't been successful and removing the comment if it wasn't, using a catch block